### PR TITLE
Apm 1276 asid required

### PR DESCRIPTION
**Pull request by Sourceless**
* Add in some templating and config for requiring (or not requiring) an ASID

### CODE REVIEW

File: proxies/live/apiproxy/targets/ig3.xml [hunk 1]
```diff
@@ -30,6 +30,16 @@
             <Step>
                 <Name>KeyValueMapOperations.GetSecureVariables</Name>
             </Step>
+            {% if REQUIRE_ASID %}
+            <Step>
+                <Name>RaiseFault.400BadRequest</Name> <!-- Might want something better here... -->
+                <Condition>(request.header.NHSD-ASID is null) or (not request.header.NHSD-ASID ~~ ".+")</Condition>
+            </Step>
+            {% endif %}
```
Comment (strutt) [inline proxies/live/apiproxy/targets/ig3.xml:38]: Swap the order of these steps

**Review by strutt — CHANGES_REQUESTED** [review]

File: proxies/live/apiproxy/targets/ig3.xml [hunk 2]
```diff
@@ -30,6 +30,16 @@
             <Step>
                 <Name>KeyValueMapOperations.GetSecureVariables</Name>
             </Step>
+            {% if REQUIRE_ASID %}
+            <Step>
+                <Name>RaiseFault.400BadRequest</Name> <!-- Might want something better here... -->
+                <Condition>(request.header.NHSD-ASID is null) or (not request.header.NHSD-ASID ~~ ".+")</Condition>
```
Comment (Sourceless) [inline proxies/live/apiproxy/targets/ig3.xml:36]: This condition needs changing the variable assigned by the next step

**Review by strutt — APPROVED** [review]

**Pull request merged**